File: src/main/kotlin/web/controller/WordController.kt
package web.controller

import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import web.service.WordService
import web.dto.Word
import javax.inject.Inject

@Controller("/words")
class WordController {

    @Inject
    lateinit var wordService: WordService

    @Get("/random{/number}")
    fun randomWords(number: Int?): List<Word>{
        return wordService.getRandomWords(number?: 1)
    }
}

File: server/src/words/word.controller.ts
import { Controller, Get, Param } from '@nestjs/common';
import { WordService } from './word.service';
import { Word } from './interfaces/word.interface';

@Controller('words')
export class WordController {
  constructor(private readonly wordService: WordService) {}

  @Get('/random/:amount')
  async random(@Param('amount') amount?: number): Promise<Word[]> {
    return this.wordService.getRandomWords(amount ? amount : 1);
  }

  @Get('/random/')
  async randomOne(): Promise<Word[]> {
    return this.wordService.getRandomWords(1);
  }
}
